---
title: "Handbook: NGS Data Analysis"
excerpt: "Handbook on conducting NGS data analysis studies. At the moment there are detailed manuals on Quality Control and Variant Calling along with the complete pipeline of simple research in phylogenetics. Much more to be disclosured soon!<br/><img src='/images/NGS workflow.png' width='500px'>"
collection: portfolio
---

Please also visit the initial [repository](https://github.com/iliapopov17/NGS-Data-Analysis-Manual). There is a guide how to install and activate the needed conda envinronment.

<img src='/images/NGS workflow.png'>

_Typical workflow of NGS data analysis_

## Much more to be disclosured soon:
- Whole Genome and Pangenome Analyses
- 16S Amplicon Analysis
- Maybe someday HiC...?

## Phylogenetics ðŸŒ³

In the [Phylogenetics folder](https://github.com/iliapopov17/NGS-Data-Analysis-Manual/tree/main/04_Phylogenetics) there is a complete pipeline of simple research in phylogenetics, from working with NCBI (and other databases) to building trees, evaluating them, and getting some worthwhile results.

## Genomic Variation Analysis ðŸ§¬

In the [Genomic Variation Analysis folder](https://github.com/iliapopov17/NGS-Data-Analysis-Manual/tree/main/02_Genomic_Variation_Analysis) there is a detailed guide how to conduct studies on Variant Calling using `fastqc`, `trimmomatic`, `bwa`, `samtools`, `abra2`, `bcftools`, `snpEff` & `SnpSift`

## Quality Control of raw data ðŸ’Ž

In the [Quality Control folder](https://github.com/iliapopov17/NGS-Data-Analysis-Manual/tree/main/01_Quality_Control) there is a detailed guide how to conduct quality control of raw data using `fastqc` and `trimmomatic`.
