---
title: "Manuals: NGS Data Analysis"
excerpt: "Manuals on conducting NGS data analysis studies. At the moment there are detailed manuals on Quality Control and Variant Calling. Much more to be disclosured soon!<br/><img src='/images/NGS analysis workflow.svg' width='500px'>"
collection: portfolio
---

Please also visit the initial [repository](https://github.com/iliapopov17/NGS-Data-Analysis-Manual). There is a guide how to install and activate the needed conda envinronment.

<img src='/images/NGS analysis workflow.svg'>

_Typical workflow of NGS data analysis_

## Much more to be disclosured soon:
- Reference-free approach in the comparative analysis of metagenomic data
- Whole Genome and Pangenome Analyses
- 16S Amplicon Analysis

## Genomic Variation Analysis

In the [Genomic Variation Analysis folder](https://github.com/iliapopov17/NGS-Data-Analysis-Manual/tree/main/2%20-%20Genomic%20Variation%20Analysis) there is a detailed guide how to conduct studies on Variant Calling using `fastqc`, `trimmomatic`, `bwa`, `samtools`, `abra2`, `bcftools`, `snpEff` & `SnpSift`

## Quality Control of raw data

In the [Quality Control folder](https://github.com/iliapopov17/NGS-Data-Analysis-Manual/tree/main/1%20-%20Quality%20Control) there is a detailed guide how to conduct quality control of raw data using `fastqc` and `trimmomatic`.
