---
title: "Detection of coronaviruses in insectivorous bats of Fore-Caucasus, 2021"
collection: publications
permalink: /publication/2023-02-09-paper-7
excerpt: 'The study focuses on the threat of Coronaviruses (CoVs), particularly those originating from bats, due to their high mutation rate. A meta-analysis found a 9.8% prevalence of CoVs in bats. The study conducted a qPCR screening of CoVs in the bat population of Fore-Caucasus and attempted to characterize them using metagenomic NGS. CoV RNA was detected in 5 samples, indicating a 3.33% prevalence in these locations. BetaCoVs reads were identified in the raw metagenomic NGS data, but detailed characterization was limited due to low RNA concentration. The study suggests a need for further research using deeper metagenomic NGS for detailed CoV characterization.'
date: 2023-02-09
venue: 'Scientific reports'
paperurl: 'http://iliapopov17.github.io/files/Papers/Detection of coronaviruses in insectivorous bats of Fore-Caucasus, 2021.pdf'
citation: 'Popov, I.V.; Ohlopkova, O.V.; Donnik, I.M.; Zolotukhin, P.V.; Umanets, A.; Golovin, S.N.; Malinovkin, A.V.; Belanova, A.A.; Lipilkin, P.V.; Lipilkina, T.A.; <b>Popov, I.V.</b>; et al. Detection of Coronaviruses in Insectivorous Bats of Fore-Caucasus, 2021. <i>Sci. Rep.</i> 2023, 13, doi:10.1038/s41598-023-29099-6.'
---

Coronaviruses (CoVs) pose a huge threat to public health as emerging viruses. Bat-borne CoVs are especially unpredictable in their evolution due to some unique features of bat physiology boosting the rate of mutations in CoVs, which is already high by itself compared to other viruses. Among bats, a meta-analysis of overall CoVs epizootiology identified a nucleic acid observed prevalence of 9.8% (95% CI 8.7–10.9%). The main objectives of our study were to conduct a qPCR screening of CoVs’ prevalence in the insectivorous bat population of Fore-Caucasus and perform their characterization based on the metagenomic NGS of samples with detected CoV RNA. According to the qPCR screening, CoV RNA was detected in 5 samples, resulting in a 3.33% (95% CI 1.1–7.6%) prevalence of CoVs in bats from these studied locations. BetaCoVs reads were identified in raw metagenomic NGS data, however, detailed characterization was not possible due to relatively low RNA concentration in samples. Our results correspond to other studies, although a lower prevalence in qPCR studies was observed compared to other regions and countries. Further studies should require deeper metagenomic NGS investigation, as a supplementary method, which will allow detailed CoV characterization.
